client/cmd/dexc/main.go

```diff
@@ -93,13 +95,17 @@ func main() {
 			rpcSrv, err := rpcserver.New(rpcCfg)
 			if err != nil {
 				log.Errorf("Error starting rpc server: %v", err)
-				os.Exit(1)
+				disconnectActive()
```

Comment:
Seem like you want to call `cancel()` here instead of calling Disconnect methods, and do not `return`.  The `cm.Wait()` call at these little goroutines will wait for the servers to return before they are allowed to return themselves.

Reply:
In the cases where there were any errors in newing  an rpc or web server or starting either the cm.wg was nil and therefore resulted in a nil pointer dereference if a return was not issued and cm.wait() was called. Please advise if the latest version with returns still in place are not the best way to handle this.

Reply:
Ah, yes that's true. `cancel()` and `return` is what you want, correct.  So to summarize the flow, `cancel` makes any other subsystem started with the `appCtx` start shutting down, while the `return` will exit the goroutine of the caller and execute `defer wg.Done()` on the way out.  I'd suggest putting a final log statement after `wg.Wait()` and `ui.Close()` so you know that it's executing all of the intended statements in `main` rather than returning to the OS the moment `os.Exit` was called as before.

Reply:
Thanks for the explanation. I've pushed an exit statement after ui.Close().